app/src/main/java/iband/com/ibandplayerdemo/PlayerActivity.java

```diff
@@ -0,0 +1,440 @@
+package iband.com.ibandplayerdemo;
+
+
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.view.WindowManager;
+import android.widget.AdapterView;
+import android.widget.ImageView;
+import android.widget.ListView;
+import android.widget.ProgressBar;
+import android.widget.SeekBar;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import com.iband.playersdk.IBandBasePlayer;
+import com.iband.playersdk.IBandError;
+import com.iband.playersdk.IBandPlayer;
+import com.iband.playersdk.IBandPlayerVariant;
+import com.iband.playersdk.IBandPlayerView;
+import com.iband.playersdk.IBandSDK;
+import com.iband.playersdk.IBandStream;
+
+import java.util.ArrayList;
+
+import iband.com.ibandplayerdemo.adapter.VariantAdapter;
+import iband.com.ibandplayerdemo.adapter.VariantHolder;
+
+public class PlayerActivity extends AppCompatActivity implements IBandPlayer.Listener, IBandStream.Listener {
+
+    private static final String STREAM_ID = "5994219bbc454f6edf961315";
+    private boolean mIsThreadPositionActive;
+    private boolean mIsUserChangePosition;
+    private int mDisplayMode;
+    private long mLastDuration = 0;
+    private long mLastPosition = 0;
+    private View mBtnBack;
+    private View mPositionActionBar;
+    private View mViewLive;
+    private View mViewDirection;
+    private View mBtnCenter;
+    private View mBottomActionBar;
+    private TextView mTvDuration;
+    private TextView mTvPosition;
+    private ImageView mBtnVariant;
+    private ImageView mBtnVrMode;
+    private ImageView mBtnVideoScale;
+    private ImageView mBtnPlayPause;
+    private ProgressBar mProgressPlayer;
+    private ProgressBar mProgressStream;
+    private ListView mLvSettings;
+    private SeekBar mSbPosition;
+    private IBandPlayerView mViewPLayer;
+    private Thread mThreadUpdatePosition;
+    private VariantAdapter mVariantAdapter;
+    private ArrayList<VariantHolder> mVariants;
+    private IBandSDK mIBandSDK;
+    private IBandPlayer mPlayer;
+    private IBandStream mStream;
+
+
+    @Override
+    public void onBackPressed() {
+        if (mViewPLayer.getVRMode()) {
+            mViewPLayer.setVRMode(false);
+            setDisplayMode();
+            return;
+        }
+        super.onBackPressed();
+    }
+
+    @Override
+    public void onPlayerStateChanged(IBandPlayer iBandPlayer, IBandBasePlayer.State state, IBandBasePlayer.State old) {
+        if (state == IBandBasePlayer.State.READY) {
+            mProgressPlayer.setVisibility(View.GONE);
```

Comment:
Move those commands to an informative function such as `initPlayerView()`

Reply:
done